README.md

```diff
@@ -83,21 +83,73 @@ sintax $(your_command). For instance:
 ```
 DATABASE=postgres://$(whoami)@localhost/database
 ```
- 
+
 Running it on the CLI:
 
 ```
 $ dotenv "echo $DATABASE"
 postgres://myusername@localhost/database
 ```
 
+### Type checking envs
+Env variables are simple strings. However, they can represent other types like
+integers, booleans, IP addresses, emails, URIs, and so on. We provide an interface
+that performs type checking after loading the envs and before running your application.
+If the type-check succeeded the application is executed, otherwise you will get an
+error with the types that mismatch.
+
+In order to use this functionality you can use the `loadSafeFile` which takes the same
+configuration value as the `loadFile` function. Also, you need to have a `.scheme.yml`
+in your current directory. This file must have the following structure:
+
+```yaml
+- type: bool
```

Comment:
@CristhianMotoche I'm not sure about this structure. Grouping variables by type look a little bit strange since it doesn't' seem very intuitive. I'm proposing to move the `type` at the `variable_name` level, something like:

```yaml
- name: PORT
  type: integer
  required: true

- name: TOKEN
  type: string
  # required: false (Make it optional, with False as default value perhaps?)
```

Then you should be able to create a data structure like:

```hs
data EnvVariable = EnvVariable
  { envVariableName      :: Text
  , envVariableType        :: EnvVariableType
  , envVariableRequired :: Bool
  }
```

Reply:
@sestrella the current structure will avoid duplication, since we will only declare the type once.
WDYT?

Reply:
@CristhianMotoche you could achieve the same using plain YAML:

```yaml
- &numeric { type: integer, required: true }

- name: HOST
  <<: *numeric

- name: DB
  <<: *numeric
```

http://yaml.org/type/merge.html

Reply:
😲 🆒 @sestrella I'll change it then.